apps/src/blockly/addons/extensions/logic_compare.ts

```diff
@@ -0,0 +1,31 @@
+import GoogleBlockly, {Block} from 'blockly/core';
+
+/** Type of a block that has LOGIC_COMPARE_ONCHANGE_MIXIN */
+type CompareBlock = Block & {
+  onchange: (this: CompareBlock) => void;
+};
+
+/**
+ * This override intentionally removes dynamic type validation for the left
+ * and right sides of a logic_compare (? == ?) block.
+ * Artist binary levels and existing student work use this block to compare
+ * characters from strings (ex. '01010101') with the numbers 0 and 1.
+ */
+const LOGIC_COMPARE_ONCHANGE_MIXIN = {
```

Comment:
will removing this check cause any issues (thinking about all our blockly-related console errors)? Or do we work around any such issues?

Reply:
I didn't know about a specific one, but I was wondering if blockly auto-disconnects them because if they stay connected something bad happens (such as a console error). If not, disregard this!

Reply:
Oh, I see!

No, nothing bad happens. We still generate valid code and everything works as far as I can tell. :)